libs/dal/src/repositories/notification-template/notification-template.repository.ts

```diff
@@ -165,15 +165,31 @@ export class NotificationTemplateRepository extends BaseRepository<
     return { totalCount: totalItemsCount, data: this.mapEntities(items) };
   }
 
-  async getList(organizationId: string, environmentId: string, skip = 0, limit = 10) {
-    const totalItemsCount = await this.count({ _environmentId: environmentId });
+  async getList(organizationId: string, environmentId: string, skip = 0, limit = 10, query?: string) {
+    let searchQuery: FilterQuery<NotificationTemplateDBModel> = {};
+    if (query) {
+      searchQuery = {
+        $or: [
+          { name: { $regex: regExpEscape(query), $options: 'i' } },
+          { 'triggers.identifier': { $regex: regExpEscape(query), $options: 'i' } },
+        ],
```

Comment:
allow searching by the name or trigger identifier case-insensitive

Reply:
Sorry, i missed this part of the search implementation.
if we want to search by both name and identifier in the same query (by OR operator), as far as I know, we need to make sure we have an index for both (all cases) meaning:
* user query by name, index `_environmentId.name`
* user query by name and identifier, index `_environmentId.identifier.name` (covers `_environmentId.identifier` and  `_environmentId.identifier.name` )

This is because if we have only `_environmentId.identifier.name` we could not use it for queries by `name` because we would be missing the `identifier` in the index.

Reply:
Ok, I do understand what you mean. I will add it, thanks! 🙌 

I see that we also have `_organizationId.triggers.identifier`, but I couldn't find any references in the code where we use `_organizationId and identifier`. Do we need it?

Also, in Atlas `_environmentId_1_triggers.identifier_1` means that we just have to extend it right?

Reply:
The only places i found that depend on `_organizationId` in their query are blueprint-related which are located in:
* NotificationTemplateRepository.findBlueprintByTriggerIdentifier
* NotificationTemplateRepository.getBlueprintList
we could easily refactor the code to use environmentId instead, however for now I would keep the index so we won't scan the collections until we refactor the code.

Thats right we could extend `_environmentId_1_triggers.identifier_1` to `_environmentId_1_triggers.identifier_1_name`